enterprise/backend/src/metabase_enterprise/task/cache.clj

```diff
@@ -4,23 +4,252 @@
    [clojurewerkz.quartzite.schedule.cron :as cron]
    [clojurewerkz.quartzite.triggers :as triggers]
    [java-time.api :as t]
-   [metabase.lib.ident :as lib.ident]
-   [metabase.public-settings.premium-features :refer [defenterprise]]
+   [metabase.models.setting :refer [defsetting]]
+   [metabase.public-settings.premium-features :as premium-features :refer [defenterprise]]
    [metabase.query-processor :as qp]
    [metabase.task :as task]
+   [metabase.util.i18n :refer [deferred-tru]]
+   [metabase.util.log :as log]
    [toucan2.core :as t2])
   (:import
+   (java.util.concurrent Callable ExecutorService Executors)
+   (org.apache.commons.lang3.concurrent BasicThreadFactory$Builder)
    (org.quartz.spi MutableTrigger)))
 
 (set! *warn-on-reflection* true)
 
+;;; ------------------------------------------- Preemptive Caching ----------------------------------------------------
+
+(defsetting preemptive-caching-thread-pool-size
+  (deferred-tru "The size of the thread pool used to preemptively re-run cached queries.")
+  :default    10
+  :export?    false
+  :type       :integer
+  :visibility :internal)
+
+(defonce ^:private pool
+  (delay (Executors/newFixedThreadPool
```

Comment:
Honestly no reason, I was copying the existing pattern in `metabase.notification.send`. Cached seems better for this since most instances won't be needing these threads constantly unless they set up a lot of preemptive caching.

Reply:
cached is unbounded correct? Fixed and bounded number of threads sounds preferable to me.

Reply:
switching to a custom `ThreadPoolExecutor` which has a max pool size